packages/actor-rdf-resolve-quad-pattern-federated/lib/FederatedQuadSource.ts

```diff
@@ -149,25 +219,28 @@ export class FederatedQuadSource implements ILazyQuadSource {
     //       See discussion at https://github.com/comunica/comunica/issues/553
     const sourcesIt = this.sources.iterator();
     const proxyIt: AsyncIterator<PromiseProxyIterator<RDF.Quad>> = sourcesIt.map((source) => {
+      const sourceId = this.getSourceId(source);
       remainingSources++;
       sourcesCount++;
 
-      // If we can predict that the given source will have no bindings for the given pattern,
-      // return an empty iterator.
-      const pattern: Algebra.Pattern = this.algebraFactory.createPattern(
-        FederatedQuadSource.nullToVariable(subject, 's'),
-        FederatedQuadSource.nullToVariable(predicate, 'p'),
-        FederatedQuadSource.nullToVariable(object, 'o'),
-        FederatedQuadSource.nullToVariable(graph, 'g'),
-      );
+      return new PromiseProxyIterator(async () => {
+        // Deskolemize terms, so we send the original blank nodes to each source.
+        const s = FederatedQuadSource.deskolemizeTerm(FederatedQuadSource.nullToVariable(subject, 's'), sourceId);
```

Comment:
No need to deskolemize thus for terms that are `null`, which seems like a common case. Perhaps `subject === null ? toVar(subject, 's') : deskolemizeTerm(subject)`?